core/chaincode/shim/chaincode.go

```diff
@@ -148,40 +147,18 @@ func getPeerAddress() string {
 	}
 
 	if peerAddress = viper.GetString("peer.address"); peerAddress == "" {
-		// Assume docker container, return well known docker host address
-		peerAddress = "172.17.42.1:30303"
+		chaincodeLogger.Fatalf("Peer address not configured, can't connect to peer")
```

Comment:
This is certainly an improvement, thank you.  Sorry to nit pick, but let me ask one question: Do you think this will sufficiently inform the user precisely what is wrong?  (I'm not saying it doesn't, I'm sincerely asking).  What I am wondering is whether referring to the configuration element more precisely may help a (perhaps novice) user know what to do.  For instance "peer.address is not configured ... " might be clearer. I can be convinced otherwise.